diff --git a/VintagestoryApi/Config/GameVersion.cs b/VintagestoryApi/Config/GameVersion.cs
index 82b1d5a..3233f0e 100644
--- a/VintagestoryApi/Config/GameVersion.cs
+++ b/VintagestoryApi/Config/GameVersion.cs
@@ -38,7 +38,7 @@ namespace Vintagestory.API.Config
         /// <summary>
         /// Assembly Info Version number in the format: major.minor.revision-[rc/pre.subrevision]
         /// </summary>
-        public const string OverallVersion = OverallMajorMinor + ".5";
+        public const string OverallVersion = OverallMajorMinor + ".6";
 
         /// <summary>
         /// Whether this is a stable or unstable version
