diff --git a/VintagestoryLib/Vintagestory.ClientNative/Screenshot.cs b/VintagestoryLib/Vintagestory.ClientNative/Screenshot.cs
index b08094b..be592aa 100644
--- a/VintagestoryLib/Vintagestory.ClientNative/Screenshot.cs
+++ b/VintagestoryLib/Vintagestory.ClientNative/Screenshot.cs
@@ -68,11 +68,11 @@ public class Screenshot
 		//IL_009b: Unknown result type (might be due to invalid IL or missing references)
 		//IL_00a1: Expected O, but got Unknown
 		//IL_00a2: Unknown result type (might be due to invalid IL or missing references)
 		//IL_00a8: Expected O, but got Unknown
 		SKBitmap val = new SKBitmap(new SKImageInfo(size.Width, size.Height, (SKColorType)6, (SKAlphaType)((!withAlpha) ? 1 : 3)));
-		GL.ReadPixels(0, 0, size.Width, size.Height, (PixelFormat)32993, (PixelType)5121, val.GetPixels());
+		Vintagestory.Client.ScreenManager.Platform.ReadDefaultFramebuffer(0, 0, size.Width, size.Height, val.GetPixels());
 		if (scaleScreenshot)
 		{
 			val = val.Resize(new SKImageInfo(((NativeWindow)d_GameWindow).ClientSize.X, ((NativeWindow)d_GameWindow).ClientSize.Y), new SKSamplingOptions(SKCubicResampler.Mitchell));
 		}
 		if (!flip)
