diff --git a/VintagestoryApi/Client/UI/Elements/Impl/Interactive/Text/GuiElementDynamicText.cs b/VintagestoryApi/Client/UI/Elements/Impl/Interactive/Text/GuiElementDynamicText.cs
index 5913ff7..734fdf3 100644
--- a/VintagestoryApi/Client/UI/Elements/Impl/Interactive/Text/GuiElementDynamicText.cs
+++ b/VintagestoryApi/Client/UI/Elements/Impl/Interactive/Text/GuiElementDynamicText.cs
@@ -68,10 +68,11 @@ namespace Vintagestory.API.Client
                 {
                     ImageSurface surface = new ImageSurface(Format.Argb32, (int)Bounds.InnerWidth, (int)Bounds.InnerHeight);
                     Context ctx = genContext(surface);
                     DrawMultilineTextAt(ctx, 0, 0, orientation);
 
+                    if (Vintagestory.API.Client.OptimumGuiGpuProbe.RenderingRequested) ctx.Dispose();
                     api.Event.EnqueueMainThreadTask(() =>
                     {
                         generateTexture(surface, ref textTexture);
                         ctx.Dispose();
                         surface.Dispose();
