diff --git a/VSSurvivalMod/Systems/SupportBeams/ModSystemSupportBeamPlacer.cs b/VSSurvivalMod/Systems/SupportBeams/ModSystemSupportBeamPlacer.cs
index 1354815..6219d49 100644
--- a/VSSurvivalMod/Systems/SupportBeams/ModSystemSupportBeamPlacer.cs
+++ b/VSSurvivalMod/Systems/SupportBeams/ModSystemSupportBeamPlacer.cs
@@ -22,8 +22,6 @@ namespace Vintagestory.GameContent
         public BlockSelection Blocksel;
         [ProtoMember(3)]
         public bool PartialEnds;
-        [ProtoMember(4)]
-        public int BlockId;
     }
 
     public class ModSystemSupportBeamPlacer : ModSystem, IRenderer
@@ -55,16 +53,29 @@ namespace Vintagestory.GameContent
         public override void StartServerSide(ICoreServerAPI api)
         {
             api.Network.GetChannel("beamplacer").SetMessageHandler<BeamPlacePacket>(OnCompletePlaceClient);
+            api.Event.PlayerDisconnect += Event_PlayerDisconnect;
             base.StartServerSide(api);
         }
 
+        private void Event_PlayerDisconnect(IServerPlayer byPlayer)
+        {
+            workspaceByPlayer.Remove(byPlayer.PlayerUID);
+        }
+
         private void OnCompletePlaceClient(IServerPlayer fromPlayer, BeamPlacePacket packet)
         {
             var ws = getWorkSpace(fromPlayer.Entity);
             fromPlayer.Entity.BlockSelection = packet.Blocksel;
             if (packet.begin)
             {
-                beginPlace(ws, api.World.Blocks[packet.BlockId], fromPlayer.Entity, packet.Blocksel, packet.PartialEnds);
+                if (packet.Blocksel?.Position == null || packet.Blocksel.Face == null || !(fromPlayer.InventoryManager.ActiveHotbarSlot.Itemstack?.Block is BlockSupportBeam))
+                {
+                    capi.Logger.Audit("An invalid support beam placement packet was sent from player {0}", fromPlayer.PlayerName);
+                }
+                else
+                {
+                    beginPlace(ws, fromPlayer.InventoryManager.ActiveHotbarSlot.Itemstack.Block, fromPlayer.Entity, packet.Blocksel, packet.PartialEnds);
+                }
             }
             else
             {
@@ -157,10 +168,9 @@ namespace Vintagestory.GameContent
             ws.block = block;
             ws.onFacing = blockSel.Face;
 
-            capi?.Network.GetChannel("beamplacer").SendPacket(new BeamPlacePacket() { 
-                Blocksel = blockSel, 
+            capi?.Network.GetChannel("beamplacer").SendPacket(new BeamPlacePacket() {
+                Blocksel = blockSel,
                 begin = true,
-                BlockId = block.Id,
                 PartialEnds = partialEnds
             });
         }
@@ -182,6 +192,24 @@ namespace Vintagestory.GameContent
                 return;
             }
 
+            if (ws.block.Id != slot.Itemstack?.Block?.Id || !(ws.block is BlockSupportBeam))
+            {
+                (api as ICoreClientAPI)?.TriggerIngameError(this, "invalbeamblock", Lang.Get("Invalid beam block"));
+                return;
+            }
+
+            var player = (byEntity as EntityPlayer)?.Player;
+            if (!api.World.Claims.TryAccess(player, ws.startPos.AddCopy(ws.startOffset), EnumBlockAccessFlags.BuildOrBreak))
+            {
+                player.InventoryManager.ActiveHotbarSlot.MarkDirty();
+                return;
+            }
+            if (!api.World.Claims.TryAccess(player, ws.startPos.AddCopy(nowEndOffset), EnumBlockAccessFlags.BuildOrBreak))
+            {
+                player.InventoryManager.ActiveHotbarSlot.MarkDirty();
+                return;
+            }
+
             if (beh == null)
             {
                 var hereBlock = api.World.BlockAccessor.GetBlock(ws.startPos);
@@ -191,13 +219,6 @@ namespace Vintagestory.GameContent
                     return;
                 }
 
-                var player = (byEntity as EntityPlayer)?.Player;
-                if (!api.World.Claims.TryAccess(player, ws.startPos, EnumBlockAccessFlags.BuildOrBreak))
-                {
-                    player.InventoryManager.ActiveHotbarSlot.MarkDirty();
-                    return;
-                }
-
                 if (eplr.Player.WorldData.CurrentGameMode != EnumGameMode.Creative)
                 {
                     int len = (int)Math.Ceiling(nowEndOffset.DistanceTo(ws.startOffset));
@@ -524,7 +545,11 @@ namespace Vintagestory.GameContent
                 if (!stable) continue;
 
                 double dist = DistanceToLine(point, beam.Start, beam.End);
-                if (dist < minDistance) minDistance = dist;
+                if (dist < minDistance)
+                {
+                    minDistance = dist;
+                    nearestBeam = beam;
+                }
             }
 
             beamstartend = nearestBeam;
