diff --git a/VintagestoryApi/Client/Render/PerceptionEffects/PerceptionEffect.cs b/VintagestoryApi/Client/Render/PerceptionEffects/PerceptionEffect.cs
index 41f461b..ac59954 100644
--- a/VintagestoryApi/Client/Render/PerceptionEffects/PerceptionEffect.cs
+++ b/VintagestoryApi/Client/Render/PerceptionEffects/PerceptionEffect.cs
@@ -1,56 +1,33 @@
-﻿using Vintagestory.API.Common;
+using Vintagestory.API.Common;
+using Vintagestory.API.MathTools;
 
-#nullable disable
+namespace Vintagestory.API.Client;
 
-namespace Vintagestory.API.Client
+public abstract class PerceptionEffect
 {
-    public abstract class PerceptionEffect
-    {
-        public string Code;
-        public int PerceptionEffectId;
-        public float Intensity;
-
-        public double DurationHours;
-
-        protected ICoreClientAPI capi;
-
-        public PerceptionEffect(ICoreClientAPI capi)
-        {
-            this.capi = capi;
-        }
-
-        public virtual void OnBeforeGameRender(float dt)
-        {
-            if (PerceptionEffectId > 0)
-            {
-                capi.Render.ShaderUniforms.PerceptionEffectId = PerceptionEffectId;
-                capi.Render.ShaderUniforms.PerceptionEffectIntensity = Intensity;
-            }
-        }
-
-        public virtual void OnOwnPlayerDataReceived(EntityPlayer eplr)
-        {
+    public string Code;
+    public int PerceptionEffectId;
+    public float Intensity;
+    public double DurationHours;
+    protected ICoreClientAPI capi;
 
-        }
-
-        public virtual void ApplyToFpHand(Matrixf modelMat)
-        {
-
-        }
-
-        public virtual void ApplyToTpPlayer(EntityPlayer entityPlr, float[] modelMatrix, float? playerIntensity = null)
-        {
-
-        }
-
-        public virtual void NowDisabled()
-        {
-            
-        }
+    public PerceptionEffect(ICoreClientAPI capi)
+    {
+        this.capi = capi;
+    }
 
-        public virtual void NowActive(float intensity)
+    public virtual void OnBeforeGameRender(float dt)
+    {
+        if (PerceptionEffectId > 0)
         {
-            this.Intensity = intensity;
+            capi.Render.ShaderUniforms.PerceptionEffectId = PerceptionEffectId;
+            capi.Render.ShaderUniforms.PerceptionEffectIntensity = Intensity;
         }
     }
+
+    public virtual void OnOwnPlayerDataReceived(EntityPlayer eplr) { }
+    public virtual void ApplyToFpHand(Matrixf modelMat) { }
+    public virtual void ApplyToTpPlayer(EntityPlayer entityPlr, float[] modelMatrix, float? playerIntensity = null) { }
+    public virtual void NowDisabled() { }
+    public virtual void NowActive(float intensity) { Intensity = intensity; }
 }
