diff --git a/VSEssentials/Loading/RegistryObjectType.cs b/VSEssentials/Loading/RegistryObjectType.cs
index 7c78fc7..fbaa92e 100644
--- a/VSEssentials/Loading/RegistryObjectType.cs
+++ b/VSEssentials/Loading/RegistryObjectType.cs
@@ -71,11 +71,11 @@ namespace Vintagestory.ServerMods.NoObf
         public RegistryObjectVariantGroup[] VariantGroups;
 
         /// <summary>
         /// Variant values as resolved from blocktype/itemtype or entitytype
         /// </summary>
-        public API.Datastructures.OrderedDictionary<string, string> Variant = new ();
+        public Vintagestory.API.Datastructures.OrderedDictionary<string, string> Variant = new ();
 
         /// <summary>
         /// (Currently unused) A set of potential world interactions for this object. Used to display what the object is used for - e.g. Shift + Right Click to Knap Stones.
         /// </summary>
         [JsonProperty]
@@ -260,16 +260,16 @@ namespace Vintagestory.ServerMods.NoObf
                 }
             }
 
         }
 
-        internal virtual RegistryObjectType CreateAndPopulate(ICoreServerAPI api, AssetLocation fullcode, JObject jobject, JsonSerializer deserializer, API.Datastructures.OrderedDictionary<string, string> variant)
+        internal virtual RegistryObjectType CreateAndPopulate(ICoreServerAPI api, AssetLocation fullcode, JObject jobject, JsonSerializer deserializer, Vintagestory.API.Datastructures.OrderedDictionary<string, string> variant)
         {
             return this;
         }
 
-        protected T CreateResolvedType<T>(ICoreServerAPI api, AssetLocation fullcode, JObject jobject, JsonSerializer deserializer, API.Datastructures.OrderedDictionary<string, string> variant) where T : RegistryObjectType, new()
+        protected T CreateResolvedType<T>(ICoreServerAPI api, AssetLocation fullcode, JObject jobject, JsonSerializer deserializer, Vintagestory.API.Datastructures.OrderedDictionary<string, string> variant) where T : RegistryObjectType, new()
         {
             T resolvedType = new T()
             {
                 Code = Code,
                 VariantGroups = VariantGroups,
@@ -301,11 +301,11 @@ namespace Vintagestory.ServerMods.NoObf
             resolvedType.Code = fullcode;
             resolvedType.jsonObject = null;
             return resolvedType;
         }
 
-        protected static void solveByType(JToken json, string codePath, API.Datastructures.OrderedDictionary<string, string> searchReplace)
+        protected static void solveByType(JToken json, string codePath, Vintagestory.API.Datastructures.OrderedDictionary<string, string> searchReplace)
         {
             if (json is JObject jsonObj)
             {
                 List<string> propertiesToRemove = null;
                 Dictionary<string, JToken> propertiesToAdd = null;
