diff --git a/Cairo/wrapper/AssemblyInfo.cs b/Cairo/wrapper/AssemblyInfo.cs
index a151668..b8bc681 100644
--- a/Cairo/wrapper/AssemblyInfo.cs
+++ b/Cairo/wrapper/AssemblyInfo.cs
@@ -1,5 +1,10 @@
 using System.Reflection;
+using System.Runtime.CompilerServices;
 
 [assembly: AssemblyVersion("1.0.0.0")]
 [assembly: AssemblyFileVersion("1.0.0.0")]
 [assembly: AssemblyDelaySign(false)]
+[assembly: InternalsVisibleTo("Optimum.Tests")]
+
+[assembly: InternalsVisibleTo("VintagestoryAPI")]
+[assembly: InternalsVisibleTo("VintagestoryLib")]
