diff --git a/VSSurvivalMod/Systems/Cooking/MealMeshCache.cs b/VSSurvivalMod/Systems/Cooking/MealMeshCache.cs
index 9a0a503..dc777da 100644
--- a/VSSurvivalMod/Systems/Cooking/MealMeshCache.cs
+++ b/VSSurvivalMod/Systems/Cooking/MealMeshCache.cs
@@ -314,11 +314,11 @@ namespace Vintagestory.GameContent
 
             if (!meshrefs.TryGetValue(mealhashcode, out MultiTextureMeshRef? mealMeshRef))
             {
                 if (Environment.CurrentManagedThreadId != RuntimeEnv.MainThreadId)
                 {
-                    if (capi != null && Environment.CurrentManagedThreadId == capi.TesselationThreadId) capi.Event.EnqueueMainThreadTask(() => GetOrCreateMealInContainerMeshRef(containerBlock, forRecipe, contentStacks, foodTranslate), "createmealmesh");    // Do it on the main thread instead
+                    if (capi != null && capi.IsTesselationThread(Environment.CurrentManagedThreadId)) capi.Event.EnqueueMainThreadTask(() => GetOrCreateMealInContainerMeshRef(containerBlock, forRecipe, contentStacks, foodTranslate), "createmealmesh");    // Do it on the main thread instead
                     throw new RetryTesselationException("Attempting to upload a new meal mesh outside of the main thread. If this occurs during tesselation, it should automatically retry. If not during tesselation, please only call this on the main thread!");
                 }
 
                 MeshData mesh = GenMealInContainerMesh(containerBlock, forRecipe, contentStacks, foodTranslate);
                 meshrefs[mealhashcode] = mealMeshRef = capi!.Render.UploadMultiTextureMesh(mesh);
