diff --git a/VintagestoryLib/Vintagestory.Client/ScreenManager.cs b/VintagestoryLib/Vintagestory.Client/ScreenManager.cs
index b5000e7..1befc0c 100644
--- a/VintagestoryLib/Vintagestory.Client/ScreenManager.cs
+++ b/VintagestoryLib/Vintagestory.Client/ScreenManager.cs
@@ -731,13 +731,14 @@ public class ScreenManager : KeyEventHandler, MouseEventHandler, NewFrameHandler
 		Platform.CheckGlError();
 		FrameProfiler.Mark("doneRender2Default");
 		Mat4f.Identity(api.renderapi.pMatrix);
 		Mat4f.Ortho(api.renderapi.pMatrix, 0f, Platform.WindowSize.Width, Platform.WindowSize.Height, 0f, 0f, 20001f);
 		Platform.GlDepthFunc(EnumDepthFunction.Lequal);
+		// Mono.Cecil transplant.
 		float num = 20000f;
-		GL.ClearBuffer((ClearBuffer)6145, 0, ref num);
-		GL.DepthRange(0f, 20000f);
+		Platform.ClearDefaultDepth(num);
+		Platform.SetDepthRange(0f, 20000f);
 		Platform.GlEnableDepthTest();
 		Platform.GlDisableCullFace();
 		Platform.GlToggleBlend(on: true);
 		CurrentScreen.RenderAfterBlit(dt);
 		if (CurrentScreen.RenderBg && !Platform.IsShuttingDown)
