diff --git a/VintagestoryApi/Client/UI/Dialog/GuiDialog.cs b/VintagestoryApi/Client/UI/Dialog/GuiDialog.cs
index 3b3daf9..daee741 100644
--- a/VintagestoryApi/Client/UI/Dialog/GuiDialog.cs
+++ b/VintagestoryApi/Client/UI/Dialog/GuiDialog.cs
@@ -573,10 +573,11 @@ namespace Vintagestory.API.Client
             }
 
             if (!IsOpened()) return;
             foreach (GuiComposer composer in composers)
             {
+                if (!composer.Enabled) continue;
                 if (composer.Bounds.PointInside(args.X, args.Y))
                 {
                     args.Handled = true;
                     return;
                 }
@@ -598,10 +599,11 @@ namespace Vintagestory.API.Client
                 if (args.Handled) return;
             }
 
             foreach (GuiComposer composer in composers)
             {
+                if (!composer.Enabled) continue;
                 if (composer.Bounds.PointInside(args.X, args.Y))
                 {
                     args.Handled = true;
                     return;
                 }
@@ -623,10 +625,11 @@ namespace Vintagestory.API.Client
                 if (args.Handled) return;
             }
 
             foreach (GuiComposer composer in composers)
             {
+                if (!composer.Enabled) continue;
                 if (composer.Bounds.PointInside(args.X, args.Y))
                 {
                     args.Handled = true;
                     break;
                 }
@@ -648,10 +651,11 @@ namespace Vintagestory.API.Client
 
             if (focused)
             {
                 foreach (GuiComposer composer in composers)
                 {
+                    if (!composer.Enabled) continue;
                     if (composer.Bounds.PointInside(capi.Input.MouseX, capi.Input.MouseY))
                     {
                         args.SetHandled(true);
                     }
                 }
