diff --git a/VSEssentials/Systems/RoomRegistry.cs b/VSEssentials/Systems/RoomRegistry.cs
index 0bb15ae..716b4f1 100644
--- a/VSEssentials/Systems/RoomRegistry.cs
+++ b/VSEssentials/Systems/RoomRegistry.cs
@@ -131,11 +131,14 @@ namespace Vintagestory.GameContent
             blockAccessor?.Dispose();
             blockAccessor = null;
 
-            foreach (var ba in disposableBlockAccessors.Values)
+            if (disposableBlockAccessors != null)
             {
-                ba?.Dispose();
+                foreach (var ba in disposableBlockAccessors.Values)
+                {
+                    ba?.Dispose();
+                }
+                disposableBlockAccessors.Clear();
             }
-            disposableBlockAccessors.Clear();
             disposableBlockAccessors = null;
         }
 
