diff --git a/VintagestoryApi/Common/Entity/EntityPlayer.cs b/VintagestoryApi/Common/Entity/EntityPlayer.cs
index 7279e48..7551c9f 100644
--- a/VintagestoryApi/Common/Entity/EntityPlayer.cs
+++ b/VintagestoryApi/Common/Entity/EntityPlayer.cs
@@ -419,20 +419,20 @@ namespace Vintagestory.API.Common
         {
             tesselating = true;
 
             base.OnTesselation(ref entityShape, shapePathForLogging);
 
-            if (Api?.Side == EnumAppSide.Client)
+            if (Api?.Side == EnumAppSide.Client && AnimManager?.Animator != null)
             {
                 AnimManager.HeadController = new PlayerHeadController(AnimManager, this, entityShape);
             }
 
             if (IsSelf)
             {
                 OtherAnimManager.LoadAnimator(World.Api, this, entityShape, OtherAnimManager.Animator?.Animations, true, null, "head");
 
-                if (Api?.Side == EnumAppSide.Client)
+                if (Api?.Side == EnumAppSide.Client && OtherAnimManager?.Animator != null)
                 {
                     OtherAnimManager.HeadController = new PlayerHeadController(OtherAnimManager, this, entityShape);
                 }
             }
         }
