diff --git a/VintagestoryLib/Vintagestory.Client.NoObf/ShaderPrograms.cs b/VintagestoryLib/Vintagestory.Client.NoObf/ShaderPrograms.cs
index f19d524..5fad806 100644
--- a/VintagestoryLib/Vintagestory.Client.NoObf/ShaderPrograms.cs
+++ b/VintagestoryLib/Vintagestory.Client.NoObf/ShaderPrograms.cs
@@ -40,10 +40,18 @@ public static class ShaderPrograms
 
 	public static ShaderProgramEntityanimated Entityanimated;
 
 	public static ShaderProgramFinal Final;
 
+	public static ShaderProgram FsrEasu;
+
+	public static ShaderProgram FsrRcas;
+
+	public static ShaderProgram TaaDebug;
+
+	public static ShaderProgram TaaResolve;
+
 	public static ShaderProgramFindbright Findbright;
 
 	public static ShaderProgramGodrays Godrays;
 
 	public static ShaderProgramGui Gui;
