diff --git a/VintagestoryLib/Vintagestory.Client.NoObf/SystemRenderFrameBufferDebug.cs b/VintagestoryLib/Vintagestory.Client.NoObf/SystemRenderFrameBufferDebug.cs
index cb0a94d..e2aac9e 100644
--- a/VintagestoryLib/Vintagestory.Client.NoObf/SystemRenderFrameBufferDebug.cs
+++ b/VintagestoryLib/Vintagestory.Client.NoObf/SystemRenderFrameBufferDebug.cs
@@ -136,54 +136,54 @@ internal class SystemRenderFrameBufferDebug : ClientSystem
 			int shadowMapQuality = ClientSettings.ShadowMapQuality;
 			if (shadowMapQuality > 0)
 			{
 				frameBufferRef = game.Platform.FrameBuffers[11];
 				debugdepthbuffer.DepthSampler2D = frameBufferRef.DepthTextureId;
-				GL.TexParameter((TextureTarget)3553, (TextureParameterName)34892, 0);
+				game.Platform.SetTextureDepthCompare(frameBufferRef.DepthTextureId, 0);
 				game.GlPushMatrix();
 				game.GlTranslate(gUIScale * 170f, gUIScale * 10f, gUIScale * 50f);
 				game.GlScale(gUIScale * 300f, gUIScale * 300f, 0.0);
 				game.GlScale(0.5, 0.5, 0.0);
 				game.GlTranslate(1.0, 1.0, 0.0);
 				game.GlRotate(180f, 1.0, 0.0, 0.0);
 				debugdepthbuffer.ProjectionMatrix = game.CurrentProjectionMatrix;
 				debugdepthbuffer.ModelViewMatrix = game.CurrentModelViewMatrix;
 				game.Platform.RenderMesh(quadModel);
 				game.GlPopMatrix();
-				GL.TexParameter((TextureTarget)3553, (TextureParameterName)34892, 34894);
+				game.Platform.SetTextureDepthCompare(frameBufferRef.DepthTextureId, 34894);
 			}
 			if (shadowMapQuality > 1)
 			{
 				frameBufferRef = game.Platform.FrameBuffers[12];
 				debugdepthbuffer.DepthSampler2D = frameBufferRef.DepthTextureId;
-				GL.TexParameter((TextureTarget)3553, (TextureParameterName)34892, 0);
+				game.Platform.SetTextureDepthCompare(frameBufferRef.DepthTextureId, 0);
 				game.GlPushMatrix();
 				game.GlTranslate(gUIScale * 170f, gUIScale * 320f, gUIScale * 50f);
 				game.GlScale(gUIScale * 300f, gUIScale * 300f, 0.0);
 				game.GlScale(0.5, 0.5, 0.0);
 				game.GlTranslate(1.0, 1.0, 0.0);
 				game.GlRotate(180f, 1.0, 0.0, 0.0);
 				debugdepthbuffer.ProjectionMatrix = game.CurrentProjectionMatrix;
 				debugdepthbuffer.ModelViewMatrix = game.CurrentModelViewMatrix;
 				game.Platform.RenderMesh(quadModel);
 				game.GlPopMatrix();
-				GL.TexParameter((TextureTarget)3553, (TextureParameterName)34892, 34894);
+				game.Platform.SetTextureDepthCompare(frameBufferRef.DepthTextureId, 34894);
 			}
 			frameBufferRef = game.Platform.FrameBuffers[5];
 			debugdepthbuffer.DepthSampler2D = frameBufferRef.DepthTextureId;
-			GL.TexParameter((TextureTarget)3553, (TextureParameterName)34892, 0);
+			game.Platform.SetTextureDepthCompare(frameBufferRef.DepthTextureId, 0);
 			game.GlPushMatrix();
 			game.GlTranslate(gUIScale * 170f, gUIScale * 630f, gUIScale * 50f);
 			game.GlScale(gUIScale * 300f, gUIScale * 300f, 0.0);
 			game.GlScale(0.5, 0.5, 0.0);
 			game.GlTranslate(1.0, 1.0, 0.0);
 			game.GlRotate(180f, 1.0, 0.0, 0.0);
 			debugdepthbuffer.ProjectionMatrix = game.CurrentProjectionMatrix;
 			debugdepthbuffer.ModelViewMatrix = game.CurrentModelViewMatrix;
 			game.Platform.RenderMesh(quadModel);
 			game.GlPopMatrix();
-			GL.TexParameter((TextureTarget)3553, (TextureParameterName)34892, 34894);
+			game.Platform.SetTextureDepthCompare(frameBufferRef.DepthTextureId, 34894);
 			debugdepthbuffer.Stop();
 			game.guiShaderProg.Use();
 			game.Platform.GlDisableDepthTest();
 			game.Render2DLoadedTexture(labels[13], gUIScale * 170f, gUIScale * 630f);
 			if (shadowMapQuality > 0)
